### PR TITLE
Apply static policy correctly to unspec

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98983

cc @soumith @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_subclasses/meta_utils.py [hunk 1]
```diff
@@ -291,7 +291,11 @@ def sym_sizes_strides_storage_offset(t):
                     from torch._dynamo.source import AttrSource
 
                     base = self.meta_tensor(
-                        t._base, shape_env, callback, source=AttrSource(source, "_base")
+                        t._base,
+                        shape_env,
+                        callback,
+                        source=AttrSource(source, "_base"),
+                        dynamic_dims=[DimDynamic.STATIC] * t._base.dim(),
```
Comment (ezyang) [inline torch/_subclasses/meta_utils.py:298]: Worth a comment here about it

File: torch/_dynamo/variables/builder.py [hunk 2]
```diff
@@ -873,11 +873,30 @@ def wrap_unspecialized_primitive(self, value):
...
 
+                name = self.source.name()
+                if name not in self.tx.output.frame_state:
+                    curr_size = value
+                else:
+                    curr_size = self.tx.output.frame_state[name]
+                    if curr_size is not None and curr_size == value:
+                        pass
+                    else:
+                        curr_size = None
```
Comment (ezyang) [inline torch/_dynamo/variables/builder.py:884]: I think we can shorten this as

<pre>
curr_size = frame_state.get(name, val)
if curr_size != val:
  curr_size = None
</pre>

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/variables/builder.py (hunk 2 above)
Comment (voznesenskym) [reply torch/_dynamo/variables/builder.py:884]: yes thats what I kept saying over VC lol. I'll rev it once I get CI signal that this is fine

File: torch/_subclasses/meta_utils.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_subclasses/meta_utils.py:298]: yes

**Comment by voznesenskym** [comment]
> CI just looks like frame counts wobbling.

I have a disabled test. 

Discussed offline, RandomSource is getting a visit.

File: test/dynamo/test_repros.py [hunk 3]
```diff
@@ -1043,11 +1043,7 @@ def test_maml_no_item_capture(self):
         for _ in range(10):
             self.assertTrue(same(opt_model(a, b, c, d), correct))
 
-        # TODO: There is some bug here where corrects ends up with
-        # a Tensor in element 0.  We graph break on that (reflected here)
-        # but really we shouldn't have gotten a Tensor at all, as
-        # the operation is between an int and an item() result
-        self.assertEqual(cnt.frame_count, ifunspec(6, 5))
+        self.assertEqual(cnt.frame_count, ifunspec(5, 5))
```
Comment (ezyang) [inline test/dynamo/test_repros.py:1046]: nit you don't need ifunspec now

**Comment by voznesenskym** [comment]
hmmm new graph breaks, will look

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**